pydantic/_internal/_config.py

```diff
@@ -257,6 +263,9 @@ def push(self, config_wrapper: ConfigWrapper | ConfigDict | None):
     json_schema_extra=None,
     strict=False,
     revalidate_instances='never',
+    val_date_or_time_unit='infer',
+    ser_date_or_time_unit='infer',
+    ser_json_datetime='iso8601',
     ser_json_timedelta='iso8601',
```

Comment:
I suppose theoretically there is a validation config here too which would make the alternative invalid, something like

```python
val_json_datetime: 'iso8601' | 'float' | 'both' = 'both'
```